File: BEmu/MarketDataRequest/ElementMarketDatetime.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu" file="MarketDataRequest/ElementMarketDatetime.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace BEmu.MarketDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class ElementMarketDatetime : Element
    {
        private readonly Datetime _datetime;
        private readonly string _name;

        internal ElementMarketDatetime(string name, DateTime datetime)
        {
            this._name = name;
            this._datetime = new Datetime(datetime, Datetime.DateTimeTypeEnum.both);
        }

        public override Name Name { get { return new Name(this._name); } }
        public override int NumValues { get { return 1; } }
        public override int NumElements { get { return 0; } }
        public override bool IsNull { get { return false; } }

        public override object this[int index]
        {
            get
            {
                if (index == 0)
                    return this._datetime;
                else
                    throw new IndexOutOfRangeException();
            }
        }

        internal override StringBuilder PrettyPrint(int tabIndent)
        {
            return base.PrettyPrintHelper(tabIndent, this._datetime.ToString());
        }

        public override object GetValue()
        {
            return this._datetime;
        }

        public override Datetime GetValueAsDatetime()
        {
            return this._datetime;
        }

        public override Datetime GetValueAsDate() //note that this does not truncate the value to date-only, it still returns both the date and the time
        {
            return this._datetime;
        }

        public override Datetime GetValueAsTime() //note that this does not truncate the value to time-only, it still returns both the date and the time
        {
            return this._datetime;
        }
    }
}

File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/MarketDataRequest/ElementMarketDatetime.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.MarketDataRequest.ElementMarketDatetime.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.MarketDataRequest;

import com.bemu.BEmu.DateTimeTypeEnum;
import com.bemu.BEmu.Datetime;
import com.bemu.BEmu.Name;

public class ElementMarketDatetime extends ElementParent
{
	private final Datetime _date;
	private final String _name;
	
	ElementMarketDatetime(String name, Datetime date)
    {
        this._date = new Datetime(date, DateTimeTypeEnum.both);
        this._name = name;
    }
	
    public Name name()
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
    	return super.prettyPrintHelper(tabIndent, this._date.toString());
    }
    
    public Object getValue()
    {
    	return this._date;
    }

    public Datetime getValueAsDatetime()
    {
        return this._date;
    }

    public Datetime getValueAsDate() //note that this does not truncate the value to date-only, it still returns both the date and the time
    {
        return this._date;
    }

    public Datetime getValueAsTime() //note that this does not truncate the value to date-only, it still returns both the date and the time
    {
        return this._date;
    }

}
